### ISSUE TITLE
Bug: lint on modified via changes in obsidian does not trigger, shows 0 characters changed

### ISSUE THREAD

**Issue by ankitson** [platers/obsidian-linter#1457]
- [x] I have verified that I am on the latest version of the Linter

## Describe the Bug

I have Linter set to lint on save, using a custom date property and parsing called "date-modified", with source of truth set to Changes in Obsidian. But saving after changing the file shows "0 characters added 0 characters removed" and the timestamp is not updated.

## How to Reproduce

Obsidian 1.11.7 on Mac OS or iOS.
Open any file with a date-modified property, add some characters, press Cmd+S to save, or open the command palette and select "Lint current file"

Config enabled: 
- lint on save, 
- display message on lint, 
- ignore files `_resources\/.*`, 
- Date Modified, 
- Date modified Key: `date-modified`
- Date modified source of truth: Changes in Obsidian
- Format: `YYYY-MM-DD hh:mm A`

Logs:
```
Running linter
rules before regular rules: 0.10000000894069672 ms
Running Custom Regex
custom regex rules: 0.4000000059604645 ms
Running YAML Timestamp
yaml-timestamp: 1 ms
---
date-created: 2026-02-02 08:25 PM
date-modified: 2026-02-08 07:17 PM
---
<note content>
rules after regular rules: 1.6000000089406967 ms
rules running: 2.4000000059604645 ms
```

## Expected Behavior

date-modified should be updated

## Screenshots

<img width="770" height="724" alt="Image" src="https://github.com/user-attachments/assets/735ed7f1-3300-4645-911c-a2e5a25e4b70" />

## Device

- [x] Desktop
- [x] Mobile

## Additional Context

This works fine with "changes in filesystem" as the modified source.

**Comment by pjkaufman** [comment]
Hey @ankitson,  if the Linter does not make changes and you change the active file, the Linter does not know that you made changes. You will need to enable the logic for running YAML Timestamp after x seconds to have it make a change to the date modified when the Linter doesn't make the change and Obsidian is the source of truth.

**Comment by ankitson** [comment]
@pjkaufman That works, thank you!

**Comment by pjkaufman** [comment]
Based on the response I got that the YAML Timestamp after x seconds setting being enabled does fix the issue, I am going to go ahead and close this since it is not really an issue and is working as designed.
